### PR TITLE
Dependency bumps, incl security bump + fluent deps

### PR DESCRIPTION
**Pull request by stevejalim**
* Resolves https://github.com/mozilla/bedrock/pull/12725
  * Security bump 
* Resolves https://github.com/mozilla/bedrock/pull/12702
  * You can check QR code generation still works at http://localhost:8000/de/firefox/107.0.0/whatsnew/
* Resolves https://github.com/mozilla/bedrock/pull/12656
  * If you make the docs, you can confirm that the fluent highlighting etc still work there
* Resolves https://github.com/mozilla/bedrock/pull/12701
* Resolves https://github.com/mozilla/bedrock/pull/12682

Integration tests pass https://gitlab.com/mozmeao/bedrock/-/pipelines/775960097

### CODE REVIEW

**Comment by stevejalim** [comment]
This PR was also going to tackle fluent-syntax, but we'll deal with that separately am still chasing down pain points in requirements compatibility

**Review by robhudson — APPROVED** [review]

File: bin/compile-requirements.sh [hunk 1]
```diff
@@ -15,6 +15,6 @@ export CUSTOM_COMPILE_COMMAND="$ make compile-requirements"
 pip install -U pip==22.3.1
 pip install pip-tools==6.10.0
 
-pip-compile --generate-hashes -r requirements/prod.in
-pip-compile --generate-hashes -r requirements/dev.in
-pip-compile --generate-hashes -r requirements/docs.in
+pip-compile --generate-hashes -r requirements/prod.in --resolver=backtracking
+pip-compile --generate-hashes -r requirements/dev.in --resolver=backtracking
+pip-compile --generate-hashes -r requirements/docs.in --resolver=backtracking
```
[the remark this answers is not in the document]
Comment (stevejalim) [reply bin/compile-requirements.sh:20]: Yeah - there's a new resolver and the old one is deprecated. This turns off the warning by switching to the new one now

File: requirements/prod.in [hunk 2]
```diff
@@ -27,7 +27,7 @@ docutils==0.18.0
 envcat==0.1.1
 everett==3.1.0
 fluent.runtime==0.3.1
-fluent.syntax==0.17.0  # hard-pinned subdep to avoid compatibility issues
+fluent.syntax==0.18.1
```
Comment (pmac) [inline requirements/prod.in:30]: glad to see work happening on Fluent. Hopefully our tests are good enough to catch any incompatibilities. We do import and use some internal classes IIRC.

Comment (stevejalim) [reply requirements/prod.in:30]: Unit and integration tests all green, at least

**Review by pmac — APPROVED** [review]
🚀

**Pull request merged**